src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Field.cs

```diff
@@ -61,7 +61,7 @@ private BoundExpression RewriteTupleFieldAccess(BoundFieldAccess node, BoundExpr
 
             // make a field access for the most local access
             WellKnownMember wellKnownTypeMember = TupleTypeSymbol.GetTupleTypeMember(currentLinkType.Arity, fieldRemainder);
-            var linkField = (FieldSymbol)Binder.GetWellKnownTypeMember(_compilation, wellKnownTypeMember, _diagnostics, syntax: node.Syntax);
+            var linkField = (FieldSymbol)Binder.GetWellKnownTypeMember(_compilation, wellKnownTypeMember, _diagnostics, type: currentLinkType.OriginalDefinition, syntax: node.Syntax);
```

Comment:
I'll discuss with @VSadov and @AlekseyTs to see if there is a way to test this now and I'll get back to you.